File: 백준/Bronze/5543. 상근날드/상근날드.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {

        int burger = 2000;
        for (int i = 0; i < 3; i++) {
            burger = Math.min(burger, Integer.parseInt(br.readLine()));
        }

        int drink = 2000;
        for (int i = 0; i < 2; i++) {
            drink = Math.min(drink, Integer.parseInt(br.readLine()));
        }

        System.out.println(burger + drink - 50);

    } // end of main

} // end of class

File: 백준/Bronze/5543. 상근날드/상근날드.py
"""
"""
import sys

input = lambda: sys.stdin.readline().strip()

burger = 2000
for _ in range(3):
    burger = min(burger, int(input()))
drink = 2000
for _ in range(2):
    drink = min(drink, int(input()))

print(burger + drink - 50)